assets/src/edit-story/components/library/panes/text/textSets/textSets.js

```diff
@@ -96,18 +104,15 @@ function TextSets() {
     [textSets]
   );
 
-  const textSetReducer = useMemo(
-    () => getTextSetReducer(textSets, allTextSets),
-    [textSets, allTextSets]
-  );
+  const rowVirtualizer = useVirtual({
+    size: Math.ceil(filteredTextSets.length / 2),
+    parentRef: paneRef,
+    estimateSize: useCallback(() => TEXT_SET_SIZE + TEXT_SET_ROW_GAP, []),
+    overscan: 5,
+  });
 
-  const [{ filteredTextSets, renderedTextSets }, dispatch] = useReducer(
-    textSetReducer,
-    {
-      filteredTextSets: [],
-      renderedTextSets: [],
-    }
-  );
+  const sectionId = useMemo(() => `section-${uuidv4()}`, []);
+  const title = useMemo(() => __('Text Sets', 'web-stories'), []);
```

Comment:
It likely doesn't buy much, but it does end up only making one function call on the first render rather than calling the `__(...args)` function every render.  I just added it in while I was memoizing the id.